codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/GoIntegration.java

```diff
@@ -49,6 +50,10 @@ default byte getOrder() {
         return 0;
     }
 
+    default ArtifactType getArtifactType() {
```

Comment:
Is it possible for a plugin to work for either artifact type? What's this used for?